src/extra/recipes/cookiebox.py

```diff
@@ -0,0 +1,1047 @@
+from typing import Optional, Union, Dict, List, Tuple, Any
+from dataclasses import dataclass, asdict, is_dataclass
+
+from .base import BaseCalibration
+
+import numpy as np
+from extra_data import open_run, by_id, DataCollection
+import h5py
+import xarray as xr
+import pandas as pd
+
+import itertools
+from concurrent.futures import ThreadPoolExecutor, ProcessPoolExecutor
+
+from extra.components import Scan, AdqRawChannel, XrayPulses, XGM
+
+from lmfit.models import GaussianModel, LinearModel, ConstantModel
+from lmfit.model import ModelResult
+import scipy
+from scipy.stats import linregress
+from scipy.optimize import minimize_scalar
+from functools import partial
+from scipy.signal import fftconvolve
+from scipy.interpolate import CubicSpline
+from scipy.signal import kaiserord, filtfilt, firwin
+
+
+@dataclass
+class TofFitResult:
+    """Class keeping track of fit in a single Tof."""
+    energy: np.ndarray
+    mu: np.ndarray
+    sigma: np.ndarray
+    A: np.ndarray
+    Aa: np.ndarray
+    offset: np.ndarray
+
+
+def search_offset(trace: np.ndarray, sigma: float=20):
+    """
+    Find highest peaks in the 1D trace.
+    """
+    axis = np.arange(len(trace))
+    gaussian = np.exp(-0.5*(axis - np.mean(axis))**2/sigma**2)
+    gaussian /= np.sum(gaussian, axis=0, keepdims=True)
+    # apply it to the data
+    smoothened = fftconvolve(trace, gaussian, mode="same", axes=0)
+    peak_idx = np.argmax(smoothened)
+    return peak_idx - 200
```

Comment:
Where is 200 coming from?

Reply:
This is giving it some slack, because we don't know how much the peak that I find is off. This is the automatic detection bit, which can fail, because the data is so noisy. If it fails, I raise an exception and write down that the user should try to figure out the parameters by themselves.

Reply:
Ok, that seems fine but I would add a comment explaining it.